### ISSUE TITLE
Performance/EndWith cop doesn't take special capture variables into account

### ISSUE THREAD

**Issue by bdewater** [rubocop-hq/rubocop#6139]
## Expected behavior

```ruby
file_name = $` if file_name =~ /\.rb\z/
``` 
not triggering the cop

## Actual behavior

A [warning](https://codeclimate.com/github/rails/rails/pull/32695): "Use `String#end_with?` instead of a regex match anchored to the end of the string."

I was able to work around it by changing it to: 
```ruby
file_name = file_name.chomp(".rb") if file_name.end_with?(".rb")
```

## Steps to reproduce the problem

Found during https://github.com/rails/rails/pull/32695 but the above should be clear enough.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by asherkach** [comment]
Is there a need for the conditional in your workaround? Does `file_name = file_name.chomp(".rb")` not also work?

**Comment by bdewater** [comment]
You're completely right! I was still thinking along the lines of the original code 🤦‍♂️

**Comment by bdewater** [comment]
Hi @Drenmi I noticed this was closed - while my Rails contribution is fixed I still believe this should be fixed in Rubocop 🙂
